hutch_python/load_conf.py

```diff
@@ -37,24 +39,34 @@ def load(filename):
     with open(filename, 'r') as f:
         conf = yaml.load(f)
     hutch_banner(conf.get('hutch', 'hutch').lower())
-    return read_conf(conf)
+    return read_conf(conf, filename)
 
 
-def read_conf(conf):
+def read_conf(conf, filename=None):
     """
     Parameters
     ----------
     conf: dict
         dict interpretation of the original yaml file
 
+    filename: str, optional
+        Path to the conf.yml file. If provided, we can include all the created
+        objects in a special hutch.db module.
+
     Returns
     ------
     objs: dict{str: object}
         Return value of load
     """
     hutch_python.clear_load()
     plugins = get_plugins(conf)
-    objects = run_plugins(plugins)
+    hutch = conf.get('hutch')
+    if filename is None:
+        objects = run_plugins(plugins)
+    elif hutch is not None:
+        conf_path = Path(filename)
+        hutch_path = conf_path.parent
+        objects = run_plugins(plugins, hutch=hutch, hutch_path=hutch_path)
```

Comment:
Might be nice to just throw a warning here on else:
```python
"Neither a configuration file or hutch was specified. Unable to load plugins"
```
Just scares me that if `hutch` is not entered properly this does nothing and does so silently